### ISSUE TITLE
ap closed after 3 minutes

### ISSUE THREAD

**Issue by Sp1r1tus** [tbnobody/OpenDTU#198]
is there a way to disable closing ap after 3 minutes? it should still alive... thx for support!

**Comment by tbnobody** [comment]
Thats currently not implemented. Will have a look at it...

**Comment by stefan123t** [comment]
@tbnobody lumapu implemented a check for the number of open connections on the APoints interface.

https://github.com/lumapu/ahoy/blob/main/tools/esp8266/ahoywifi.cpp#L81-L85
```c
                uint8_t cnt = WiFi.softAPgetStationNum();
                if(cnt > 0) {
                    DPRINTLN(DBG_INFO, String(cnt) + F(" client connected, resetting AP timeout"));
                    mNextTryTs = (millis() + (WIFI_AP_ACTIVE_TIME * 1000));
                }
```

**Comment by madmartin** [comment]
Before implementing a logic like "keep AP open as long as there is a connected client" without other logic changes will create a big security whole!
Every time the ESP is restartet there is now an AP with a "googelable" default password (which will be in databases of hacking tools soon). When this AP opens, my evil neighbor can connect and do something.... When this connection can be kept open (in parallel to the "usual" connection from OpenDTU as WiFi-Client to my WiFi-Router) this will be definitely not something I want.

**Comment by stefan123t** [comment]
@madmartin this is already the case if it remains as it is:
> Every time the ESP is restarted there is *already* an AP with a "googelable" default password
> (which will be in databases of hacking tools soon). 
> When this AP opens, my evil neighbor can connect and do something...

Maybe you want to add an option to change the default AP password too ?

**Comment by madmartin** [comment]
Yes of course, I already mentioned that somehere here earlier.
IMHO the AP should be opened ONLY for the first time configuration, so maybe unless the WiFi-STA parameters have been set and successfully connected. 
For a later reconfiguration, if you need an AP again, there should be a special procedure like a double reset or 6x reset (like tasmota does) within a short time.

**Comment by stefan123t** [comment]
@tbnobody did you already fix the AP password pre-requisite from @madmartin ?
The feature that @Sp1r1tus requested so the AP will stay open for another minute in case someone is still connected could then be safely implemented, IMHO. 

Though I do not share the same security consciuousness as @madmartin, someone indeed could run a WiFi DeAuth Attack on my ESP based OpenDTU and try to steal / eavesdrop my WLAN Password or my start to remote control my Solar inverter.
As the communication between DTU and Inverter is insecure by design I do not think this is a real necessity / pre-requisite for this feature request.

**Comment by Sp1r1tus** [comment]
Maybe it is better to change minutes in same web gui like password change?

**Comment by stefan123t** [comment]
@tbnobody did you add the "keep AP open when clients are connected" feature ?

**Comment by tbnobody** [comment]
AP timeout can be configured now. Implemented in 8bfa7e530cfdd1bab1455c523ebb0a66a93c8b20
